### PR TITLE
Refactor player base interactions into an interface

### PR DESCRIPTION
**Pull request by Remscar**
Reorganized any non weapon base files into a `demo_player` folder.
Refactored any code that touches the player pawn to do so through an `ISWBPlayer` interface so the player code doesn't code smell into the player code too much.

### CODE REVIEW

File: code/swb_base/ui/SniperScope.cs
Comment (timmybo5) [inline code/swb_base/ui/SniperScope.cs:40]: Why do you check for both here, wouldn't the ISWBPlayer check be enough?

File: code/ExamplePlayer.cs
Comment (timmybo5) [inline code/ExamplePlayer.cs:68]: private field name should be lower camel case

File: code/ExamplePlayer.cs
Comment (timmybo5) [inline code/ExamplePlayer.cs:89]: Not a big fan of moving these out of the weapon base, could you explain your reasoning for this?

File: code/demo_player/editor/ui/AttachmentEditorMenu.razor
Comment (timmybo5) [inline code/demo_player/editor/ui/AttachmentEditorMenu.razor:2]: I was thinking `swb_player` might be a better name since most consumers will use the player and base together.

File: code/swb_base/util/ScreenUtil.cs
Comment (timmybo5) [inline code/swb_base/util/ScreenUtil.cs:54]: This should check for a player specifically, not ModelEntity to make it more efficient.

File: code/swb_base/util/ScreenUtil.cs
Comment (timmybo5) [inline code/swb_base/util/ScreenUtil.cs:68]: I don't like obj being used as a variable name for a player because it should be clear what data the variable references.

File: code/swb_base/ui/SniperScope.cs
Comment (timmybo5) [inline code/swb_base/ui/SniperScope.cs:52]: Why not use the player variable here instead of ent?

File: code/swb_base/ui/SniperScope.cs
Comment (timmybo5) [inline code/swb_base/ui/SniperScope.cs:69]: Why not use the player variable here instead of ent?

**Review by timmybo5 — CHANGES_REQUESTED** [review]
Great work on the PR! I wasn't sure yet how to best take care of the obsolete folder player files but this solution works nicely creating an abstract layer between the weapon base and player base allowing for a better and easier implementation by the consumers and their own player base. 

I've left some minor comments and will do a second review after you take care of these :)

File: code/swb_base/ui/SniperScope.cs
Comment (Remscar) [reply code/swb_base/ui/SniperScope.cs:40]: This can be fixed by making `ISWBPlayer` extend from `IEntity` which I will do.

File: code/ExamplePlayer.cs
Comment (Remscar) [reply code/ExamplePlayer.cs:89]: If this isn't moved out of the weapon base then you are forcing consumers to use the `FirstPersonCamera` and `ThirdPersonCamera` classes which come with SWB. They might have a different camera system and don't want to use your camera. I think the weapon base should be as limited as possible to weapons. What if the game that's trying to use SWB is an isometric camera system, or third person only, or maybe a something entirely different?

File: code/demo_player/editor/ui/AttachmentEditorMenu.razor
Comment (Remscar) [reply code/demo_player/editor/ui/AttachmentEditorMenu.razor:2]: I will rename the folder

File: code/swb_base/util/ScreenUtil.cs
Comment (Remscar) [reply code/swb_base/util/ScreenUtil.cs:54]: `ISWBPlayer` doesn't have `PhysicsBody` as a member of it, and since there is no `IModelEntity` it becomes a slippery slope. I think my ideal here would be to refactor out the screen shake code from swb_base and move it into the example player code since screen shake effects the camera and therefore should be handled by the consumer code.

Alternatively / in addition, could we use `Entity.Position` instead of `ModelEntity.PhysicsBody.MassCenter` instead?

File: code/swb_base/util/ScreenUtil.cs
Comment (Remscar) [reply code/swb_base/util/ScreenUtil.cs:68]: Fixed to use the type checked `ModelEntity ply` while waiting for response to the other comment regarding this function.

File: code/swb_base/ui/SniperScope.cs
Comment (Remscar) [reply code/swb_base/ui/SniperScope.cs]: Same as above; `ISWBPlayer` didn't derive from `IEntity` but it should so I can use `ply` now instead.

File: code/swb_base/ui/SniperScope.cs
Comment (Remscar) [reply code/swb_base/ui/SniperScope.cs]: Same as above; ISWBPlayer didn't derive from IEntity but it should so I can use ply now instead.

File: code/ExamplePlayer.cs
Comment (timmybo5) [reply code/ExamplePlayer.cs]: That's def something to revisit later on, as for the FOV modifiers they are already customizable and can be fully disabled.

**Comment by timmybo5** [comment]
I see you moved a lot of other stuff to the new player folder, could you explain your reason behind that:

1.  `Commands.cs`: Contains commands that control attachments which do not depend on the player. 
2.  The premade weapon UI that can be disabled with the weapon settings
3.  The editor which is used to set ironsight/run/inspect, and attachment positions.

**Comment by Remscar** [comment]
> I see you moved a lot of other stuff to the new player folder, could you explain your reason behind that:
> 
> 1. `Commands.cs`: Contains commands that control attachments which do not depend on the player.
> 2. The premade weapon UI that can be disabled with the weapon settings
> 3. The editor which is used to set ironsight/run/inspect, and attachment positions.

1. All of those commands either are debug/test commands or are for attachment editing. Not all games will want to allow players to edit gun models, attachments, or to add attachments to guns. There is no access control code in those files either so how would I prevent a player from accessing them when I don't want them to. Additionally a couple depend on the player, `PlayerBase` implements `ToggleModelEditor/ToggleAttachmentEditor`
2. Those settings are controlled through console commands that should really not be accessible to players in some games. Including the UI in the weapon base just means more headache for the consumer to figure out how to disable it, and then force it to remain disabled. I'm seeing now there's more code in the weapon base relating to the UI which I believe should also get refactored out but i'm hesitant to do in this PR since it's already quite large.
3. I don't think editor code should be included in final game code, we go back to the problem of preventing players from accessing features they should use. I should launch the SWB Base Demo to edit stuff, or another I include an additional folder from SWB Base that enables editting.

File: code/swb_base/ISWBPlayer.cs
Comment (timmybo5) [inline code/swb_base/ISWBPlayer.cs:39]: `IsLookingAtGun` could be confused with inspecting, I'd suggest renaming this to `IsLookingAtGun`

File: code/ExampleCommands.cs
Comment (timmybo5) [inline code/ExampleCommands.cs:67]: Could we cast to the player interface here and move the command to `swb_base/Commands.cs`, the test commands are really just for testing in the demo gamemode but attachments are part of the base, inseparable.

**Review by timmybo5 — CHANGES_REQUESTED** [review]

File: code/ExampleCommands.cs
Comment (Remscar) [reply code/ExampleCommands.cs:67]: But that means that by using the weapon base, anybody can just call `swb_attachment_equip` and attach an attachment. What if I don't want players to be able to equip attachments? What if they need to buy them or something.

This would force me to then go into `swb_base/Commands.cs` and make modifications to enforce access to attachments.

File: code/swb_base/ISWBPlayer.cs
Comment (Remscar) [reply code/swb_base/ISWBPlayer.cs:39]: Renamed

File: code/ExampleCommands.cs
Comment (timmybo5) [reply code/ExampleCommands.cs:67]: I will think of a solution to this for a future update, there should be a way to disable that.

**Review by timmybo5 — APPROVED** [review]
Great work, lets finally merge these changes!

**Pull request merged**
